File: socket-app/src/main/java/com/socket/utils/SessionManager.java
package com.socket.utils;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

import com.socket.handler.ClientHandler;

public class SessionManager {

    private final Map<Long, ClientHandler> sessions;

    public SessionManager() {
        this.sessions = Collections.synchronizedMap(new HashMap<Long,ClientHandler>());
    }

    public void registerSession(Long sessionId, ClientHandler handler) {
        sessions.put(sessionId, handler);
    }

    public void unregisterSession(Long sessionId) {
        sessions.remove(sessionId);
    }

    public Integer getTotalActiveSessions() {
        return sessions.size();
    }

}

File: socket-app-python/utils/session_manager.py
class SessionManager:
    
    def __init__(self):
        self.sessions = {}

    def register_session(self, session_id, handler):
        self.sessions[session_id] = handler

    def unregister_session(self, session_id):
        if session_id in self.sessions:
            del self.sessions[session_id]

    def get_total_active_sessions(self):
        return len(self.sessions)
